specs/capella/light-client/full-node.md

```diff
@@ -0,0 +1,74 @@
+# Capella Light Client -- Full Node
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Helper functions](#helper-functions)
+  - [`compute_merkle_proof_for_block_body`](#compute_merkle_proof_for_block_body)
+  - [Modified `block_to_light_client_header`](#modified-block_to_light_client_header)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This upgrade adds information about the execution payload to light client data as part of the Capella upgrade.
+
+## Helper functions
+
+### `compute_merkle_proof_for_block_body`
+
+```python
+def compute_merkle_proof_for_block_body(body: BeaconBlockBody,
+                                        index: GeneralizedIndex) -> Sequence[Bytes32]:
+    ...
+```
+
+### Modified `block_to_light_client_header`
+
+```python
+def block_to_light_client_header(block: SignedBeaconBlock) -> LightClientHeader:
+    epoch = compute_epoch_at_slot(block.message.slot)
+
+    if epoch >= CAPELLA_FORK_EPOCH:
```

Comment:
Is it because `block` could be `bellatrix.SignedBeaconBlock` here? What do you think about adding a docstring about it?

Reply:
Yes, exactly (or, in theory, even all the way back to Altair or Phase0, if bellatrix / altair never finalized, and finalized header is still back in those ancient forks).

Added a comment to clarify that.